src/js_realm.cpp

```diff
@@ -249,11 +249,28 @@ JSValueRef RealmGetProperty(JSContextRef ctx, JSObjectRef object, JSStringRef pr
     return NULL;
 }
 
+std::string RealmObjectTypeForValue(SharedRealm &realm, JSContextRef ctx, JSValueRef value) {
+    if (JSValueIsObject(ctx, value) && JSObjectIsConstructor(ctx, (JSObjectRef)value)) {
```

Comment:
When we parse the schema we accept non-constructor functions which have a schema. Here we only accept constructors. Which is correct? The docs should probably explain what types of functions users can pass in...